packages/core/content-manager/admin/src/preview/pages/Preview.tsx

```diff
@@ -87,6 +89,32 @@ const AnimatedArrow = styled(ArrowLineLeft)<{ $isSideEditorOpen: boolean }>`
   transition: rotate 0.2s ease-in-out;
 `;
 
+type MessageType =
+  | (typeof INTERNAL_EVENTS)[keyof typeof INTERNAL_EVENTS]
+  | (typeof PUBLIC_EVENTS)[keyof typeof PUBLIC_EVENTS];
+
+/**
+ * A function factory so we can generate a new sendMessage everytime we need one.
+ * We can't store and reuse a single sendMessage because it needs to have a stable identity
```

Comment:
Won't it have a stable identity if it's defined outside the react component? Otherwise wouldn't the useEffect complain the getSendMessage is not in its dep array?

I don't really have any issue with any of this just curious...